com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -938,6 +938,31 @@ SkinnedMeshRenderer unitySkin
 
                 // add bind pose
                 ExportBindPose (unitySkin, fbxNode, fbxScene, skinnedMeshToBonesMap);
+
+                // now that the skin and bindpose are set, make sure that each of the bones
+                // is set to its original position
+                var bones = unitySkin.bones;
+                foreach (var bone in bones)
+                {
+                    var fbxBone = MapUnityObjectToFbxNode[bone.gameObject];
+                    ExportTransform(bone, fbxBone, newCenter: Vector3.zero, TransformExportType.Local);
+
+                    // Cancel out the pre-rotation from the exported rotation
+
+                    // Get prerotation
+                    var fbxPreRotationEuler = fbxBone.GetPreRotation(FbxNode.EPivotSet.eSourcePivot);
+                    // Convert the prerotation to a Quaternion
+                    var fbxPreRotationQuaternion = ModelExporter.EulerToQuaternion(fbxPreRotationEuler);
+                    // Inverse of the prerotation
+                    fbxPreRotationQuaternion.Inverse();
+
+                    // Multiply LclRotation by pre-rotation inverse to get the LclRotation without pre-rotation applied
+                    var fbxFinalQuat = fbxPreRotationQuaternion * EulerToQuaternion(new FbxVector4(fbxBone.LclRotation.Get()));
+
+                    var quatToEulerMatrix = new FbxAMatrix();
+                    quatToEulerMatrix.SetQ(fbxFinalQuat);
+                    fbxBone.LclRotation.Set(ToFbxDouble3(quatToEulerMatrix.GetR()));
```

Comment:
does this take into account rotation order?

Reply:
By this point both the prerotation and local rotation used in the calculations have already been converted to Maya coords with XYZ rotation order. FBXAMatrix will return an euler in XYZ rotation order as well, so everything should already be in the right coordinates and rotation order.